### ISSUE TITLE
[string-strip-html] URLs are output even if tag is stripped

### ISSUE THREAD

**Issue by cossssmin** [codsen/codsen#54]
**Package's name**

string-strip-html

**Describe the bug**

```js
const html = `
  Show in plaintext
  <not-plaintext>
    <p>No images? <a href="url">Click here</a>.</p>  
  </not-plaintext>
`

stripHtml(`a <pre><code>void a;</code></pre> b`, {
  stripTogetherWithTheirContents: ['not-plaintext'],
  dumpLinkHrefsNearby: {
    enabled: true
  }
})
```

Result:

```html
Show in plaintext
url
```

Expected:

```html
Show in plaintext
```

**Comment by revelt** [comment]
hi Cosmin! Sorry about that, good spot, it is a bug.

**Comment by cossssmin** [comment]
Hey Roy, thanks! If you can patch it, is it possible to patch the latest CommonJS version too, please?

**Comment by revelt** [comment]
Release incoming (ESM). The CJS is tricky; it's stuck on 8.3.0 and to release let's say 8.4.0 I'd have to create a separate repo like I did for email-comb (https://github.com/codsen/email-comb-noesm) and release from there. I can see [50x more consumption](https://www.npmjs.com/package/string-strip-html) (see versions tab) of CJS compared to ESM. I'm on my lunch so CJS `8.4.0` will come sometime tonight or weekend worst case.

**Comment by cossssmin** [comment]
Thanks Roy! Yeah, Maizzle alone is responsible for ~20k downloads/month for CJS :)

By the way, I think this would be a patch release right? So it would be `8.3.1` and not `8.4.0` (unless you're planning on adding some features to `8.x`, of course).

**Comment by revelt** [comment]
Hm, we'd be porting `11.6.9` to 8.x so at least minor would make sense. We can't do major, technically that would be the right, but I checked the changelog, all differences were cosmetic and actually patch-level, corrections to whitespace processing, doctype reporting in callbacks etc.

**Comment by revelt** [comment]
PS. Github webhooks are [degraded](https://www.githubstatus.com), so no ESM release until it's sorted, CI didn't fire.

**Comment by revelt** [comment]
ESM v[11.6.10](https://www.npmjs.com/package/string-strip-html/v/11.6.10) released; CJS tbc

**Comment by cossssmin** [comment]
Hi Roy, any news on the CJS version? 😇

**Comment by revelt** [comment]
hi Cosmin! If all goes well, I aim today; if not, tomorrow. Off-topic, I was thinking, I could help to look to migrate maizzle to ESM, we will have to migrate it sooner or later.

**Comment by revelt** [comment]
Cool! There's nice pattern I found in Remix, they are not on ESM but need to consume ESM packages, so they import ESM packages asynchronously from an otherwise CJS environment, see https://remix.run/docs/en/v1/guides/mdx#advanced-configuration — they import ESM's asynchronously, with `await Promise.all`

**Comment by cossssmin** [comment]
Yeah, that's the 'easy' part. Haven't found a good way of [bypassing the cache](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c#how-can-i-import-a-module-and-bypass-the-cache-for-testing), though.

**Comment by revelt** [comment]
btw first stage is complete, I patched all known `string-strip-html` issues/features/bugs, the [11.6.13](https://www.npmjs.com/package/string-strip-html/v/11.6.13) is out, now onto CJS
